Dockerfile

```diff
@@ -16,7 +16,7 @@ FROM base AS version
 ARG PKG=github.com/distribution/distribution/v3
 RUN --mount=target=. \
   VERSION=$(git describe --match 'v[0-9]*' --dirty='.m' --always --tags) REVISION=$(git rev-parse HEAD)$(if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi); \
-  echo "-X ${PKG}/version.Version=${VERSION#v} -X ${PKG}/version.Revision=${REVISION} -X ${PKG}/version.Package=${PKG}" | tee /tmp/.ldflags; \
+  echo "-X ${PKG}/version.version=${VERSION#v} -X ${PKG}/version.revision=${REVISION} -X ${PKG}/version.mainpkg=${PKG}" | tee /tmp/.ldflags; \
```

Comment:
Wait, does that work with non-exported fields? TIL; thought they had to be exported for the `-X` functions to work, but perhaps that's just what I'm used to 😂

We should start looking if we can use go's built-in metadata for this (although I think that's producing versions based on the module version, not specifically git state, and we'd probably still want to have options to override these)

Reply:
See the comment in @corhere's original PR